x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -114,6 +116,15 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
     }
   }, [appTitle, appId]);
 
+  useEffect(() => {
+    if (state.createCaseFlyout.isSubmit) {
+      setValue((prev) => ({
+        ...prev,
+        isSubmit: true,
+      }));
+    }
+  }, [state.createCaseFlyout.isSubmit]);
+
```

Comment:
Probably the main Cases context is not a good place to store this `isSubmit` flag, this context stores general information for all cases components, and this `isSubmit` prop is too specific, it is needed only in the createCaseFlyout when the tour is open, we should keep this context independent from components logic.

There's another way to do it, we can inject the `state.createCaseFlyout.isSubmit` to the getCreateCaseFlyoutLazyNoProvider props: 
```
getCreateCaseFlyoutLazyNoProvider({
  ...state.createCaseFlyout.props,
  autoSubmit: state.createCaseFlyout.autoSubmit,
})
```
at https://github.com/elastic/kibana/blob/69c3244606ff20546608f7b9dff5dd448f35ad36/x-pack/plugins/cases/public/components/cases_context/cases_global_components.tsx#L17

And wire everything down to the `CreateCaseFormFields` component, which will receive this flag by props.

Besides that, I would also change the name of this state prop. `isSubmit` makes sense for things that can *be* "submit", like buttons. But what we are trying to do is not that, we are creating a prop that automatically submits the form when is true, so I think `autoSubmit` makes more sense and makes it easier to understand for an unfamiliar eye.